webextension/scripts/settings.js

```diff
@@ -88,6 +128,52 @@ function selectall () {
   }
 }
 
+function validatePrivateMode (event) {
+  let checkboxes = $('[name="private-inlude"]')
+  let checkedCount = checkboxes.filter((_index, item) => item.checked === true).length
+  if (checkedCount > 0) {
+    $('#private-mode').prop('checked', false)
+  } else {
+    $('#private-mode').prop('checked', true)
+  }
+
+  // Checking previous state when the only one item is checked and remove 'selected-prior' class from everywhere
+  if (checkedCount === 0 && $('.selected-prior').length === 1) {
+    $('.selected-prior').removeClass('selected-prior')
+  }
+
+  // If the event.taget.checked is true,then check if it the checkbox is only item, then remove 'selected-prior' from all
+  // And add class 'selected-prior' to the event.taget, if it is NOT there
+  // Else If the event.taget.checked is false, check if the check-box is NOT the only item, then remove class 'selected-prior'
+  if (event.target.checked === true) {
+    if (checkedCount === 1) {
+      $('.selected-prior').removeClass('selected-prior')
+    }
+    if (!$(event.target).hasClass('selected-prior')) {
+      $(event.target).addClass('selected-prior')
+    }
+  } else if (event.target.checked === false) {
+    if (!(checkedCount === 0)) {
+      $(event.target).removeClass('selected-prior')
+    } else {
+      if (!$(event.target).hasClass('selected-prior')) {
+        $(event.target).addClass('selected-prior')
+      }
+    }
+  }
+}
+
+function togglePrivateMode () {
+  if ($('#private-mode').is(':checked')) {
+    chrome.runtime.sendMessage({ message: 'clearCountBadge' })
+    chrome.runtime.sendMessage({ message: 'clearCountCache' })
```

Comment:
Do we need these? When I turn Private mode ON, the toolbar state and badge count are cleared even if we remove these lines.
Also, if we are clearing the Badge count and Toolbar state without needing to reload the popup then we need to hide blue buttons and "Saved X times" text too. It already works fine when I reload the popup, because the WB Count and Archived Resources options aren't selected.

Maybe we should do somehting like this:
`if ($('#private-mode').is(':checked')) {`
    `$('#wayback-count-label').hide()`
    `$('#borrow_books').hide()`
    `$('#news_recommend').hide()`
    `$('#wikibooks').hide()`
    `$('#doi').hide()`
  `}`

And, when I turn private mode OFF after reloading the popup, then reload the page, it doesn't shows 'R' and 'WB Count'. This is because it doesn't restores the previous selections after popup reload.

Reply:
Should we disable/enable the last save, when Private mode is ON/OFF, without needing to reload the popup?

Reply:
@tikhsuP Thanks for some good testing. Please test with the new changes I have pushed and let me know how I can improve it better. I am able to restore the previous state even if the popup is closed